src/vehicle_gui.cpp

```diff
@@ -774,45 +766,24 @@ struct RefitWindow : public Window {
 	{
 		uint row = 0;
 
-		for (uint i = 0; i < NUM_CARGO; i++) {
-			for (uint j = 0; j < this->list[i].size(); j++) {
-				const RefitOption &refit = this->list[i][j];
-
-				/* Hide subtypes if sel[0] does not match */
-				if (this->sel[0] != (int)i && refit.subtype != 0xFF) continue;
-
+		for (const auto &pair : refit_list) {
+			for (const RefitOption &refit : pair.second) {
 				if (row == click_row) {
-					this->sel[0] = i;
-					this->sel[1] = j;
+					this->selected_refit = &refit;
```

Comment:
This feels icky to me. Maybe I'm overlooking something, but now  `this->selected_refit` is a pointer to something in `refit_list`. So anything that touched `this->refit_list` should ensure `this->selected_refit` remains valid (or at best gets set to `nullptr`. Given that `BuildRefitList` clears it, doesn't `OnInvalidateData`'s call to `BuildRefitList` cause its subsequent call to `RefreshScrollbar` to read the memory that got freed by `BuildRefitList`?

Reply:
Ok... `selected_refit` is now always revalidated in `BuildRefitList()`. This means the separate code path in `OnInit()` is not necessary.

Alternatively I could keep a copy of the selected item instead of a pointer to it.